src/main/java/CheeseBot.java

```diff
@@ -0,0 +1,372 @@
+    private static int numberOfTasks = 0;
+    private static final Task[] TASKS_LIST = new Task[100];
+
+
+    public static void listTasks() {
+        if (isTasksListEmpty()) {
+            return;
+        }
+        printTasksList();
+    }
+
+    private static void printTasksList() {
+        boolean isAllTasksDone = true;
+        System.out.println("\tHere's your current list of tasks:");
+
+        for (int i = 0; i < numberOfTasks; i++) {
+            Task currentTask = TASKS_LIST[i];
+            if (!currentTask.isTaskDone()) {
+                isAllTasksDone = false;
+            }
+            System.out.println("\t" + (i + 1) + "." + currentTask);
+        }
+        System.out.println("\tNow you have " + numberOfTasks + " tasks in your list.");
+
+        if (isAllTasksDone) {
+            System.out.println("\tExcellent! You have completed all your tasks!");
+        }
+        System.out.println("\t-------------------------------------------------------------------");
+    }
+
+    public static void addTask(String[] arguments) {
+        String command = arguments[0];
+        String taskName = arguments[1];
+
+        switch (command) {
+        case "todo":
+            TASKS_LIST[numberOfTasks] = new Todo(taskName);
+            break;
+        case "deadline":
+            String by = arguments[2];
+            TASKS_LIST[numberOfTasks] = new Deadline(taskName, by);
+            break;
+        case "event":
+            String start = arguments[2];
+            String end = arguments[3];
+            TASKS_LIST[numberOfTasks] = new Event(taskName, start, end);
+            break;
+        }
+        numberOfTasks++;
+        System.out.println("\tYou have added: " + taskName);
+        System.out.println("\tYou have a total of " + numberOfTasks + " completed and uncompleted tasks.");
+        System.out.println("\t-------------------------------------------------------------------");
+    }
+
+    public static boolean isValidInput(String input) {
+        if (input.isEmpty()) {
+            System.out.println("\tInput is empty! Please type something.");
+            return false;
+        }
+
+        int spaceIndex = input.indexOf(" ");
+        if (spaceIndex == -1) {
+            return isValidSingleWordCommand(input);
+        }
+
+        String command = input.substring(0, spaceIndex);
+        switch (command) {
+        case "todo":
+            return true;
+
+        case "deadline":
+            return isValidDeadlineInput(input);
+
+        case "event":
+            return isValidEventInput(input);
+
+        case "mark":
+            //Fallthrough
+        case "unmark":
+            return isValidMarkAndUnmarkInput(input, spaceIndex);
+
+        default: // case where there is a space but command does not match any
+            System.out.println("\tNo such command!");
+            return false;
+        }
+    }
+
+    private static boolean isValidSingleWordCommand(String input) {
```

Comment:
perhaps can rewrite to camelCase for methods

Reply:
sorry what do you mean by this? is my method not in camelCase?

Reply:
No worries happy cny😆